### PR TITLE
Homework-Assignment-React-Tech-It-Easy

### PR DESCRIPTION
**Pull request by mjerta**
[update]

Second part of this particular assignment finished. I made it and  also experimented a bit with some of the syntanx and try to focus on reusability.
I learn a thing or two about the ternary operator and figured out I could even use expressions inside of an object. This way I could iterate whitout really depending on which function I really need. So there is no need to use a if statement for it.
As I said I went a bit overboard. I could have perhaps just use some functions for the buttons. But this way I really could experiment on what I could do just with the  object of e.target. Also I could as many buttons I want to it based on what 
I would put in the object of the constant of sort-implements.js

Like wise for the second reader  have a great time reading the code

Greetings,

Maarten

Dear Reader,

It's fun to get started with React. I must say it has been quite doable until now.
I noticed that the first time I worked on this assignment, I already achieved some nice results, but later on, I refactored it to make it a bit more maintainable.

It was fun to tinker around with it.
I also added some responsiveness to the styling just for fun.

I hope you have a great time reviewing my code.

Greetings,

Maarten

- **Add Opdracht 1 - Helperfuncties**
- **Added result from opdracht 2**
- **Add all styling**
- **Refactor names and exports**
- **Forgot to refactor the name of nameOfTelevision**
- **Remove all imports from helper functions**
- **Refactor some helper names to more general names so they could be used more generally in the future**
- **Add buttons with an console log**
- **Remove index.css that is not neeeded**
- **Clean code in the App.css**

### CODE REVIEW

**Comment by smallproject** [comment]
Hey Maarten,

Ik heb er wat meegekeken. In het algemeen het ziet wel goed uit.

Ik heb er weinig toe te voegen.

Goed bezig

File: src/App.jsx [hunk 1]
```diff
@@ -1,8 +1,156 @@
...
+  console.log(getListSuitableForSport(inventory));
+  console.log(getTelevionsAboveSize(inventory, 65));
+  console.log(getTelivisionsWithAmbiLight(inventory));
+  console.log("/////////////// End of examples assignment1 ///////////////\n\n")
+
+  const [sortedInventory, setSortedInventory] = useState(inventory);
+
+  function handleClick(e) {
+    const sortkey = e.target.name;
+    const result = [...inventory].sort(sortImplements[sortkey].sortFunction)
```
Comment (S-Barnhoorn) [inline src/App.jsx:58]: Leuk gedaan!

File: src/App.jsx [hunk 2]
```diff
@@ -1,8 +1,156 @@
...
 function App() {
+  console.log(getAllTvTypes(inventory));
+  console.log(getAllSoldOutTv(inventory, productOutOfStock));
+  console.log(getSpecificitem(inventory, "NH3216SMART"));
+  console.log(getListSuitableForSport(inventory));
+  console.log(getTelevionsAboveSize(inventory, 65));
+  console.log(getTelivisionsWithAmbiLight(inventory));
+  console.log("/////////////// End of examples assignment1 ///////////////\n\n")
+
+  const [sortedInventory, setSortedInventory] = useState(inventory);
```
Comment (S-Barnhoorn) [inline src/App.jsx:54]: Nu al begonnen met state! Lekker bezig hoor.

File: src/components/box.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      <p>{title}</p>
+      <p>{value}</p>
+    </div>
+  )
+
+}
+
+export default Box;
+
+Box.propTypes = {
```
Comment (S-Barnhoorn) [inline src/components/box.jsx:15]: propTypes is niet persé noodzakelijk omdat de data allemaal intern binnen de applicatie zit. Wanneer de data vanuit een API naar binnen wordt gehaald is het gebruik van propTypes wel interessant omdat je zo mogelijke problemen kan opvangen wanneer bijvoorbeeld de data die je uit de API haalt is veranderd.

**Review by S-Barnhoorn — CHANGES_REQUESTED** [review]
Goed bezig Maarten! Je bent ook goed jezelf aan het uitdagen en aan het experimenteren.

Let wel op de naming conventions van files. Components en Pages dienen altijd te beginnen met een hoofdletter. Voor helper functies of andere niet component files mag je kleine letters gebruiken.

**Pull request merged**